compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -647,16 +715,39 @@ impl Context {
             predicate_index
         };
 
-        let resolved_array = dfg.resolve(array);
-        let map_array = last_array_uses.get(&resolved_array) == Some(&instruction);
+        Ok((new_index, new_value))
+    }
 
-        if let Some(new_value) = new_value {
-            self.array_set(instruction, new_index, new_value, dfg, map_array)?;
-        } else {
-            self.array_get(instruction, array, new_index, dfg)?;
-        }
+    fn convert_array_set_store_value(
+        &mut self,
+        store_value: &AcirValue,
+        dummy: AcirVar,
+    ) -> Result<AcirValue, RuntimeError> {
+        match store_value {
+            AcirValue::Var(store_var, _) => {
+                let true_pred =
+                    self.acir_context.mul_var(*store_var, self.current_side_effects_enabled_var)?;
+                let one = self.acir_context.add_constant(FieldElement::one());
+                let not_pred =
+                    self.acir_context.sub_var(one, self.current_side_effects_enabled_var)?;
+                let false_pred = self.acir_context.mul_var(not_pred, dummy)?;
+                // predicate*value + (1-predicate)*dummy
+                let new_value = self.acir_context.add_var(true_pred, false_pred)?;
+                Ok(AcirValue::Var(new_value, AcirType::field()))
+            }
+            AcirValue::Array(values) => {
+                let mut elements = im::Vector::new();
 
-        Ok(())
+                for val in values {
+                    elements.push_back(self.convert_array_set_store_value(val, dummy)?);
```

Comment:
You need to compute the dummy here, which is different for each element.. so you also need the index